packages/cozy-harvest-lib/src/components/TriggerManager.jsx

```diff
@@ -104,57 +96,75 @@ export class TriggerManager extends Component {
    */
   async handleAccountUpdateSuccess(account) {
     this.setState({
-      updatedAccount: account
+      account
     })
 
     const { trigger } = this.props
     return await this.launch(trigger)
   }
 
+  async handleSubmit(data) {
+    const { konnector, saveAccount } = this.props
+
+    const { account } = this.state
+    const isUpdate = !!account
+
+    this.setState({
+      status: RUNNING
+    })
+
+    const savedAccount = accounts.mergeAuth(
+      await saveAccount(
+        konnector,
+        isUpdate
+          ? accounts.mergeAuth(account, data)
+          : accounts.build(konnector, data)
+      ),
+      data
+    )
+
+    return isUpdate
+      ? this.handleAccountUpdateSuccess(savedAccount)
+      : this.handleAccountCreationSuccess(savedAccount)
+  }
+
   /**
    * Launches a trigger
    * @param  {Object}  trigger io.cozy.triggers document
    * @return {Promise}         [description]
    */
   async launch(trigger) {
-    const { launchTrigger, onLoginSuccess, waitForLoginSuccess } = this.props
+    const {
+      launchTrigger,
+      onLoginSuccess,
+      onSuccess,
+      waitForLoginSuccess
+    } = this.props
 
     const job = await waitForLoginSuccess(await launchTrigger(trigger))
 
+    this.setState({
+      status: IDLE
+    })
+
     if (['queued', 'running'].includes(job.state)) {
-      onLoginSuccess(trigger)
-      this.setState({
-        status: LOGGED
-      })
+      return onLoginSuccess(trigger)
     }
 
-    this.setState({
-      status: SUCCESS
-    })
+    return onSuccess(trigger)
   }
 
   render() {
-    const { account, konnector, onDone, running } = this.props
-    const { createdAccount, status, updatedAccount } = this.state
-    const succeed = createdAccount && [LOGGED, SUCCESS].includes(status)
```

Comment:
No more LOGGED or SUCCESS status ? Is this done higher up ?

Reply:
Nope. The triggerManager is either `IDLE` or `RUNNING`.

The success is handler higher up, via `onLoginSuccess` and `onSuccess` hooks.